source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -541,12 +541,12 @@ Clients MUST ignore the response to the hello or legacy hello command when measu
 Errors encountered when running a hello or legacy hello command MUST NOT update the topology.
 (See `Why don't clients mark a server unknown when an RTT command fails?`_)
 
-Clients MUST use RTT samples to calculate an approximation for the 90th
-percentile RTT for each server using the `t-digest algorithm`_.
+Clients MUST track the minimum RTT out of the (at most) last 10 samples. Clients
+MUST report the minimum RTT as 0 until at least 2 samples have been gathered.
 
 When constructing a ServerDescription from a streaming hello or legacy hello response,
-clients MUST use average and 90th percentile round trip times from the RTT
-task.
+clients MUST set the average and minimum round trip times from the RTT task as the
+"roundTripTime" and "minRoundTripTime" fields, respectively.
```

Comment:
This requirement has actually lead to significant confusion when implementing features in the Go driver that depend on up-to-date avg/min RTT values. Using values for average and minimum RTT pulled from server descriptions can lead to difficult-to-diagnose bugs unless devs realize that the RTT values on server descriptions are never updated after handshake. The scope for Go Driver 2.0 currently includes removing all values from server descriptions that are not derived directly from the connection/handshake process. See [GODRIVER-2691](https://jira.mongodb.org/browse/GODRIVER-2691) for more details.

Should we remove this requirement and the corresponding `roundTripTime` and `minRoundTripTime` fields from a `ServerDescription`?

P.S. I realize this is only updating the existing requirement, but this could be a good time to reconsider the need for these fields.

Reply:
I don't agree because even fields that are reported by the server in the hello command response can become stale. For example, hosts, isWritablePrimary, $clusterTime, operationTime, etc.. In fact, almost every single field can become stale right after the server sends it.